src/Views/JobsView.vala

```diff
@@ -99,4 +98,39 @@ public class Printers.JobsView : Gtk.Frame {
             row1.set_header (null);
         }
     }
+
+    public void clear_queue () {
+        var dialog = new ClearQueueDialog (printer) {
+            transient_for = (Gtk.Window) get_toplevel ()
+        };
+        dialog.response.connect ((response_id) => {
+            dialog.destroy ();
+
+            if (response_id == Gtk.ResponseType.OK) {
+                list_box.@foreach ((row) => {
+                    var job = ((JobRow)row).job;
+                    job.purge ();
+                });
+
+                refresh_job_list ();
+            }
+        });
+
+        dialog.present ();
+    }
+
+    private void refresh_job_list () {
+        list_box.@foreach ((row) => {
+            list_box.remove (row);
+        });
+
+        var jobs = printer.get_jobs (true, CUPS.WhichJobs.ALL);
+        foreach (var job in jobs) {
+            if (!job.canceled_or_aborted) {
```

Comment:
Hm. I dunno I think that's debatable. It might be valuable to still be able to see these jobs. Since this isn't related to clearing the queue can we revisit this in another branch?

Reply:
The problem with that is that pressing "Clear Queue" does not then result in a "No pending jobs" view which is what I would expect.  Without the "Clear Queue" button this was not such an issue.

Reply:
I'm not sure what you mean. Can you describe steps to reproduce? In my testing here, all jobs are cleared from the queue including canceled jobs

Reply:
The problem is, this PR clears the queue by cancelling remaining jobs, because as discussed above there is no CUPS function to actually clear the data. So either we can relabel the button "Clear unfinished Jobs" or try to call an external function to actually clear the canceled and aborted jobs as well.

Reply:
Also, the linked issues are motivated by privacy concerns so leaving the canceled jobs visible did not make sense.

Reply:
Hmm, strange - I tested again without the filter and you are right the canceled jobs now permanently disappear.  I am sure that did not happen when I originally wrote the code otherwise I would not have put the filter in :shrug:  I have noticed before that CUPS does not always behave as expected.  I'll take the filter out then as it appears that CUPS *should* purge the data anyway.

Reply:
My guess is that may have been fixed along the way with some of the other branches we've merged :shrug: